src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -484,6 +484,44 @@ namespace shogun
 			    ->add_col_vec(A, i, b, result, alpha, beta);
 		}
 
+		/** Calculates pseudo inverse A+ from the singular value
```

Comment:
This doc is not entirely correct, as this is the self adjoint version, it uses an Eigensolver if the provided psd matrix, not an SVD.

As suggested before, I would put a flag in `self_adjoint` that defaults to false.
This would then use the SVD to compute the singular values (the rest of the code is identical for non-self-adjoint).

If we dont do this, there will be lots of redundant code (basically copy paste for the non self-adjoint version)

Reply:
(and then rename it to pinv)

Reply:
Actually, let's keep both methods in.
So instead of the flag, lets add another method called `pinv`
But make both methods share code via a helper method, they only compute the decomposition themselves